doc/workshops/scientific-computing.rst

```diff
@@ -2,4 +2,760 @@
+
+What is the conclusion of all this? There seems to be little or no evidence in our data to think that there is indeed a correlation between a person's pet preferences and his or her ability to solve practical or abstract problems. Cat and dog owners, rejoice!
+
+
+
+================
+Report and share
+================
+
+In science, sharing results, good and bad, is critical. Today, in scientific computing, it is also crucial to share data, and the code used to process and analyze it. To help with this, Spyder has two tools: Code Analysis and spyder-notebook.
+
+
+Code Analysis
+~~~~~~~~~~~~~
+
+When you share code, you want it to be readable, clean, and not overly complex. The Code Analysis component can help us detect these issues and even bugs that can affect the performance of our code.
+
+To see an example, you can open the ``spyder_workshop_2_astro.py`` file and run the Code Analysis (open the file in the pane and click on the green triangular button in the upper right corner).
+
+.. image:: images/workshop-2/code-analysis-pane-workshop-2.png
+   :alt: Code Analysis pane Example for astrological sign and cognitive ability
+
+You can see four categories that can help you improve your code:
+
+* Convention: programming standard violations
+* Refactor: refactoring related checks
+* Warning: Python-specific problems
+* Error: probable bugs in the code
+
+Each of these categories indicates the type of alert and the line on which the potential issue occurs. For example, it tells us that line 129 is too long, or that we imported numpy but did not use it at all in the code.
+
+These suggestions can contribute to sharing clean code. We recommend using this panel to polish your code before publishing it.
+   
+
+
+===========
+Final words
+===========
+
+In this workshop we have taken the first steps to use Spyder for scientific computing. We have seen how from a question or doubt a hypothesis is raised. This question can also emerge from the exploration of some data. This exploration can be done by calculating some measures (such as mean, median, standard deviation) or by drawing some plots.
+
+Often, in order to answer the question, we must process some data (for example, for the construction of a cognitive ability test).
+
+Finally, we try to establish a relationship between variables with some statistical tests. The results of these tests will support or not our initial hypothesis.
+
+Spyder has many features that can help you do data analysis. You can find more information in our `official documentation <https://docs.spyder-ide.org/5/index.html>`_.
+
+Also, if you are interested in an introduction to financial analysis with Spyder, you can visit the workshop :doc:`Financial Data Analysis with Spyder<../workshops/financial>`.
```

Comment:
```suggestion
Also, if you are interested in an introduction to financial analysis with Spyder, you can visit the workshop :doc:`Financial Data Analysis with Spyder<../workshops/financial>`.
```
I would put this at the end of this section since this takes them away from this workshop.